web/styles/popovers.css

```diff
@@ -752,6 +752,11 @@ ul {
 }
 
 .emoji-popover-root {
+    /* Fixes incorrect emoji popover rendering on small screens in edit mode.  */
+    [data-reference-hidden] {
+        display: inherit;
+    }
```

Comment:
Can you explain the interaction this addresses in more detail?  I'd like this comment to explain the problem enough details (like what CSS this overrides) that one wouldn't need to re-debug to figure out what this does and if/when it could be removed if we make other changes.

Reply:
Yeah, sure. This definitely needs more details to be written. Does something like this sound good?:
```javascript
/* By default, the `[data-reference-hidden]` attribute has a `display: none` value.
In this case, the emoji popover doesn't show in edit mode when we try to render it as a
centered overlay, because Tippy sets this attribute to the popover. Since we don't need
to hide this emoji popover when the reference is hidden, this solution is chosen. */

```

Reply:
In general, I have a feeling that with this code, we're solving the effect and not the cause and there may be other solutions I haven't found yet.

@amanagr, maybe you could look at this? You have experience with `[data-reference-hidden]` and the problems around that.

Reply:
This seems like the correct solution for this.

Reply:
That comment is better but to be honest I still don't understand it. The type of comment I'd want is (I'm guessing base done the name what to say -- no idea if this is correct):

"[data-reference-hidden] is set by Tippy when the element the popover is based on is hidden. By default, it has `display: none`, in order to automatically have the popover disappear if a rerender causes the element that this was based on to no longer exists.

For the emoji picker when editing a previously sent message message, this is problematic for mobile size windows because <explanation of the situation where it's bad>, so we disable this Tippy feature for that popover. We avoid having the popover remain visible if when the element element the popover was pointed to was moved by <whatever this PR does to avoid that>"

(Writing this down makes me worried that this code could introduce a bug where if someone deletes a message you've opened the reaction picker for, we'll be left with an orphaned emoji picker)

Reply:
My suspicion about this solution was confirmed. I debugged this with a clear head and realized that for mobiles, when using the newly created centered overlay API, we set the reference with `top: 0; left: 0` - which is out of the viewport. Tippy considered this as a hidden reference and hid the popover.

I removed this CSS fix and added the solution in this commit: [6a5dfd75](https://github.com/zulip/zulip/pull/26354/commits/6a5dfd75bab8d6e473ddc1ca9e8b4ddd615acb38). I'm not sure if the implementation is good enough, so I put it in a separate commit, which we could squash later.

Reply:
OK, this makes sense, squashed!